Ship_Game/GameScreens/FleetDesign/FleetDesignScreen_Draw.cs

```diff
@@ -355,8 +356,7 @@ void DrawSelectedData(SpriteBatch batch, DrawTimes elapsed)
                 RequisitionForces.Visible = true;
                 SaveDesign.Visible = true;
                 LoadDesign.Visible = true;
-                if (f.Ships.Count > 0 )
-                    AutoArrange.Visible = true;
+                AutoArrange.Visible = f.DataNodes.Where(n => n.Ship == null && n.Goal == null).Count() == 0;
```

Comment:
```suggestion
     AutoArrange.Visible = !f.DataNodes.Any(n => n.Ship == null && n.Goal == null);
```

Try avoid using LINQ. it is slow as F*** ,  we have faster implementations for these.   its not so severe in the UI, but lets keep it ordered. :)